SourceX/miniwin/misc.cpp

```diff
@@ -322,11 +335,21 @@ HWND CreateWindowExA(
     HINSTANCE hInstance,
     LPVOID lpParam)
 {
+#ifndef SWITCH
 	if (SDL_Init(SDL_INIT_EVERYTHING & ~SDL_INIT_HAPTIC) <= -1) {
 		SDL_Log(SDL_GetError());
 		return NULL;
 	}
 	atexit(SDL_Quit);
+#else
+	if (SDL_Init(SDL_INIT_VIDEO | SDL_INIT_JOYSTICK | SDL_INIT_GAMECONTROLLER) <= -1) {
+		SDL_Log(SDL_GetError());
+		return NULL;
+	}
+#endif
+
+	SDL_JoystickOpen(0);
+	SDL_GameControllerOpen(0);

```

Comment:
Missing error handeling

Reply:
Fixed in 
https://github.com/diasurgical/devilutionX/pull/201/commits/13b16680628ae5da138176c9dbea6623641aa399

in PR #201